checking

alex-belto opened the pull request.

alex-belto committed:
made client entity

alex-belto committed:
made controllers:
- getMenu
- add and remove item from order

alex-belto committed:
made payment process

alex-belto committed:
move Factory dir to src

alex-belto committed:
changed payment strategy logic,
add tips logic

alex-belto committed:
remove all logic from controller to services,
remove Dish/Drink entity
add new field - type to MenuItem

alex-belto committed:
Merge pull request #3 from alex-belto/client

Client

alex-belto committed:
Merge branch 'main' into Fixtures

alex-belto committed:
update menu fixtures

alex-belto committed:
Merge pull request #4 from alex-belto/MenuFixtures

Menu fixtures

alex-belto committed:
changed order logic
add waiter logic

alex-belto committed:
add repository

alex-belto committed:
add restaurant work logic

alex-belto committed:
Merge pull request #5 from alex-belto/waiter

Waiter

alex-belto committed:
add listener logic
change client/waiter logic
add kitchener logic

alex-belto committed:
Merge pull request #6 from alex-belto/kitchener

add listener logic/ kitchener logic

alex-belto committed:
add kitchener repository

alex-belto committed:
Merge pull request #7 from alex-belto/kitchener

add kitchener repository

alex-belto committed:
make restaurant work with order logic

alex-belto committed:
add staff fixtures
change staff logic

alex-belto committed:
change payment strategy

alex-belto committed:
change payment strategy

alex-belto committed:
create restaurant entity

alex-belto committed:
add menu items field to Restaurant

alex-belto committed:
add new fields to order: price & tips
add tips strategy

alex-belto committed:
add tips strategy
change payment methods

alex-belto committed:
Merge pull request #8 from alex-belto/restaurant

Restaurant

alex-belto committed:
restaurant manager:
add restaurant staff
add restaurant menuItems
visitors circling

alex-belto committed:
restaurant manager:
change request

alex-belto committed:
restaurant manager:
change request

alex-belto committed:
restaurant & client tables drop logic

alex-belto committed:
made restaurant a singleton
made a method to count how much order with tips was

alex-belto committed:
fixes

alex-belto committed:
listener fixes

alex-belto committed:
delete checker

alex-belto committed:
payment bug fix

alex-belto committed:
payment bug fix

alex-belto committed:
bug fixing,
di

alex-belto committed:
fix circle construct problem

alex-belto committed:
move payment and make_orders to listeners

alex-belto committed:
add calculation clients with tips

alex-belto committed:
add remove func for orders and clients

alex-belto committed:
add remove restaurant logic

alex-belto committed:
Merge pull request #9 from alex-belto/restaurant_manager

Restaurant manager

alex-belto committed:
rename classes,
add comments to each class,
drop staff tips after close restaurant

alex-belto committed:
Merge pull request #11 from alex-belto/naming

rename classes, add comments...

vitaliysheverov commented:
Спробуй обдумати архітектуру, багато менеджерів, дійсно потрібні речі краще виділити в більш конкретні класи, а решту спробувати перенести в Entity

alex-belto committed:
fix - Order завжди містить яку суму чайових, що не зовсім правильно згідно вимог. Чайових може й не бути

alex-belto committed:
fix - краще уточнити що це (randomStaff) буде індекс, $randomStaffIdx
      Окрім цього $staffs, так як це колекція

alex-belto committed:
fix - class потрібно вказувати лише якщо вище вказаний id сервісу, а не власне його назва

alex-belto committed:
fix - перейменувати методи так, щоб було зрозуміло, що саме вони роблять, а не коли

alex-belto committed:
fix - Чи може ресторан існувати без заповненої проперті $days?

alex-belto committed:
fix - Таке повідомлення кінцевому користувачу нічого не розкаже, проте Restaurant closed! чітко дає зрозуміти
 $message = "file not found!";

alex-belto committed:
fix - Таке повідомлення кінцевому користувачу нічого не розкаже, проте Restaurant closed! чітко дає зрозуміти
 $message = "file not found!";

alex-belto committed:
fix - Використати константу
 private int $status = 1;

alex-belto committed:
fix - Константа не використана
 private ?int $tipsStrategy = 1;

alex-belto committed:
fix - єдиного стилю дотримуватися, в більшості з малої, отже й тут з малої букви починати

$this->waiters = new ArrayCollection();
$this->Kitcheners = new ArrayCollection();

alex-belto committed:
fix - константи

$tipsStrategy = match ($restaurant->getTipsStrategy()) {
1 => $this->tipsStandardStrategy,

alex-belto committed:
fix - remove extra logic

alex-belto committed:
fix - change class name

alex-belto committed:
fix - updated class comments

alex-belto committed:
fix - deleted redundant logic

alex-belto committed:
fix - got rid of additional property

alex-belto committed:
fix - по замовчуванню буде витягувати по id, тому варто просто find!
return $this->restaurantRepository->findOneBy(['id' => $restaurantId]);

alex-belto committed:
fix - function hireStaff
спрости метод, зробивши 2 із 1

alex-belto committed:
fix - rename services property name according to service name

alex-belto committed:
fix - rename services name from 'manager' to better names

alex-belto committed:
Merge pull request #13 from alex-belto/bug_fixing

Bug fixing

vitaliysheverov commented on src/Services/Tips/TipsDistributor.php:
```diff
@@ -27,29 +21,26 @@ class TipsManager
+    public function splitTips(Order $order): void
     {
         $restaurant = $order->getWaiter()->getRestaurant();
         $tipsStrategy = match ($restaurant->getTipsStrategy()) {
```
Варто відловити помилку гарно, якщо $restaurant-> getTipsStrategy() поверне неіснуючу стратегію

vitaliysheverov commented on src/DataFixtures/StaffFixtures.php:
```diff
@@ -2,35 +2,35 @@
     {
         for ($i = 1; $i <= 7; $i++) {
-            $this->staffManager->createStaff('waiter');
+            $this->staffResolver->createStaff('waiter');
```
Resolver це щось більше складніше ніж просто створення, у фіксутури це буде ж завжди створення

vitaliysheverov commented on src/Entity/Kitchener.php:
```diff
@@ -9,7 +9,7 @@
 use Doctrine\ORM\Mapping as ORM;
 
 /**
- * The "Kitchener" entity stores information about chefs, including their orders,
```
З таким описом Entity мала б називатися Chef, тут два варіанти, або опис підкорегувати, або саму ентіті, репозиторій і тд

vitaliysheverov commented on src/Entity/MenuItem.php:
```diff
@@ -6,7 +6,7 @@
 
 /**
- * The "MenuItem" entity stores information about menu items, including their name,
+ * Stores information about menu items, including their name,
```
Entity це не колекція, це щось одне, тому можна трішки підкорегувати
Represents menu item, including its name, preparation time, price, associated restaurant, and related order. 
Варто переглянути по всіх Entity

vitaliysheverov commented on src/EventListener/Order/OrderListener.php:
```diff
@@ -3,28 +3,35 @@
-    public function __construct(PaymentManager $payOrder) {
-        $this->payOrder = $payOrder;
+    /**
+     * @param PaymentHandler $paymentHandler
```
Лишній @param PaymentHandler $paymentHandler

vitaliysheverov commented on src/EventListener/Order/OrderListener.php:
```diff
@@ -3,28 +3,35 @@
             $client = $order->getClient();
-            $order->setTips(rand(5, 20));
-            $this->payOrder->payOrder($client);
+            $order->setTips(rand(0, 20));
```
памятай про правило як найменше коду в if, якщо це можливо

vitaliysheverov commented on src/Repository/KitchenerRepository.php:
```diff
@@ -40,7 +40,7 @@ public function remove(Kitchener $entity, bool $flush = false): void
     }
 
 //    /**
-//     * @return KitchenerManager[] Returns an array of KitchenerManager objects
```
Лишні коментарі вкінці мають бути видалені

vitaliysheverov commented on src/Services/Client/ClientManager.php:
```diff
@@ -12,7 +12,7 @@
 
 /**
- * The class is responsible for creating clients and managing their orders.
+ * Responsible for creating clients and managing their orders.
```
A manager is not responsible for creating itself

vitaliysheverov commented on src/Services/Payment/CardPaymentProcessor.php:
```diff
@@ -47,10 +47,9 @@ public function __construct(
     public function pay(Client $client, Order $order): void
     {
         if ($this->cardValidation->isCardValid($client)) {
```
спрости if

vitaliysheverov commented on src/Services/Payment/OrderValue.php:
```diff
@@ -7,7 +7,7 @@
+ * Calculates the total cost of an order,
  * including tips, and checks if the client has sufficient funds.
  */
 class OrderValue extends AbstractController
```
чому сервіс, з назвою як у Value Object, розширяє AbstractController

vitaliysheverov commented on src/Services/Payment/PaymentHandler.php:
```diff
@@ -75,11 +75,11 @@ public function payOrder(Client $client): void
             case 'cash':
-                $paymentStrategy = $this->cashPayment;
+                $paymentStrategy = $this->cashPaymentProcessor;
                 $isEnoughMoney = $this->orderValue->isEnoughMoney($client);
```
я не можу зрозуміти, що таке orderValue тут, а це означає, що назва скоріш за все підібрана неправильно
Дублювання коду $isEnoughMoney = $this->orderValue->isEnoughMoney($client);

alex-belto committed:
fix - Чому типи подано через анотації, а не через стандартний тайпхінтинг. php 8.2 дозволяє додати тип property.

about sugar - i read, but not sure if i wanna use it

alex-belto committed:
fix -
Захардокджений шлях, який працюватиме лише для твоєї системи, з твоїми назвами та шляхами. Зокрема в мене шлях var/wwww/app може відрізнятися.
Краще винести в конфіг файли з можливістю оприділяти в .env

alex-belto committed:
fix -
Чому усі проперті nullable, проперті мають мати такий ж тип як в базі даних, тобто якщо запису без id не існує, то й тут він не може бути nullable.

alex-belto committed:
fix -
used 'return_early' rule for 'if'

alex-belto committed:
fix -
памятай про правило як найменше коду в if, якщо це можливо

alex-belto committed:
fix -
used return first principe to if

alex-belto committed:
fix -
change vision for a few methods

alex-belto committed:
fix -
 divided builder class on a few smaller classes

alex-belto committed:
fix -
 rename methods in listeners

alex-belto committed:
fix -
 remove not needed extensions

alex-belto committed:
fix -
 remove staff creation to factory

alex-belto committed:
fix -
 Removed potentially duplicate code

alex-belto committed:
fix -
 add default rule for tips distributor in a case it will get wrong tips strategy

alex-belto committed:
fix -
 changed kitchener entity comment

alex-belto committed:
fix -
 changed menuItem entity comment

alex-belto committed:
fix -
 removed unnecessary comments

alex-belto committed:
fix -
 removed param annotations

alex-belto committed:
fix -
 removed factory method from manager class

alex-belto committed:
fix -
 removed persist from builder

alex-belto committed:
fix -
 debuger

alex-belto committed:
fix -
 remove order value method
 add order value methods to Order & Client entities
 fixed bug with tips excessive withdrawal of money
 fixed bug with counting amount of clients with tips

alex-belto committed:
fix -
 add function which calculate max visitors per day for restaurant

alex-belto committed:
fix -
 removed a possible bug with double withdrawal from the card

alex-belto committed:
fix -
 removed connection MenuItem - Order
 made connection manyToMany for MenuItem - Order through OrderItem Entity

alex-belto committed:
fix -
 move delete methods from repository

alex-belto committed:
Merge pull request #14 from alex-belto/check_branch_fix

Check branch fix

alex-belto committed:
made payment transaction and add new custom exception

alex-belto committed:
- remove persist from factories
- rewrite menuItem factory
- rewrite fixtures

alex-belto committed:
Merge pull request #15 from alex-belto/check_branch_fix

- remove persist from factories

vitaliysheverov commented on src/Services/Staff/StaffFactory.php:
```diff
@@ -0,0 +1,28 @@
+<?php
```
well done

alex-belto committed:
install php unit

alex-belto committed:
- moved cardValidation service to client entity
- fixed problem with card validation exception
- moved catch cardValidException from CardPaymentProcessor to Payment Handler method

alex-belto committed:
Merge pull request #16 from alex-belto/transactions

Transactions

alex-belto committed:
Merge pull request #17 from alex-belto/unitTestInstall

install php unit

alex-belto committed:
- updated phpunit xml file
- write test for CardPaymentProcessorTest service

alex-belto committed:
- updated test for CardPaymentProcessorTest service
   -added few new test for exceptions

alex-belto committed:
made tests:
   - for CashPaymentProcessorTest
   - for PaymentHandler
moved is enoughMoney method from payment strategy to paymentHandler

alex-belto committed:
fix paymentHandler money bug

alex-belto committed:
made test for TipsCardPaymentDecorator

alex-belto committed:
made test for TipsCashPaymentDecorator

alex-belto committed:
move unit tests in unit directory
change doctrine exception to php exception in PaymentHandler.php and PaymentHandlerTest.php

alex-belto committed:
used services tags to provide them to payment handler and ClientFactory.php
removed unused elements

alex-belto committed:
tests codestyle

alex-belto committed:
rename test

alex-belto committed:
used testCase setUp
and removed mockBuilders

alex-belto committed:
changed access area for a few properties in MenuItem.php

alex-belto committed:
restaurant builder refactor

alex-belto committed:
Merge pull request #19 from alex-belto/pr_12_fix

Pr 12 fix

alex-belto committed:
Merge branch 'main' into unit_tests

# Conflicts:
#	src/Services/Restaurant/RestaurantBuilder.php

alex-belto committed:
restaurant provider test

alex-belto committed:
Merge pull request #20 from alex-belto/unit_test_1

restaurantProvider test

alex-belto committed:
restaurantProvider test upd

alex-belto committed:
restaurantProvider test upd

alex-belto committed:
restaurantManager test upd

alex-belto committed:
create restaurantBuilder test

alex-belto committed:
create ClientFactoryTest

alex-belto committed:
remove service client manager to listener clientListener
create ClientListenerTest

alex-belto committed:
create OrderListenerTest

alex-belto committed:
create OrderCleanerTest

alex-belto committed:
create MenuFixturesTest.php

alex-belto committed:
limited the length of string fields

alex-belto committed:
checked and updated the type for get and set methods

alex-belto committed:
Replaced the previous data source with a new one
Symfony\Component\DependencyInjection\ServiceLocator;
Modified tests to ensure compatibility with the changes

alex-belto committed:
Replaced the payOrder method from Payment service to Entity Client
Modified tests to ensure compatibility with the changes

alex-belto committed:
Removed extra parameter from PaymentHandler.php
Modified tests to ensure compatibility with the changes
led to a single style code all tests

alex-belto committed:
modify CashPaymentProcessorTest.php

alex-belto committed:
change return type in CardPaymentProcessorTest.php & CashPaymentProcessorTest.php
change code style in TipsCardPaymentDecoratorTest.php & TipsCashPaymentDecoratorTest.php

alex-belto committed:
add test exception case for CashPaymentProcessorTest.php

alex-belto committed:
add few new tests to Payment Handler &&
and deleted extra test dependent on paymentStrategies

alex-belto committed:
rebuild client entity tests with data provider

alex-belto committed:
used array_fill to fillUp arrayCollection in ClientListenerTest.php

alex-belto committed:
Merge pull request #18 from alex-belto/unit_tests

Unit tests

alex-belto committed:
create RestaurantManagerControllerTest.php

alex-belto committed:
upload test library
change order getTips field
create RestaurantManagerControllerTest.php

alex-belto committed:
minor fixes for ClientListener.php
minor fixes for RestaurantProvider.php
upd RestaurantManagerControllerTest.php
upd RestaurantProviderTest.php

vitaliysheverov commented on src/Services/Cleaner/OrderCleaner.php:
```diff
@@ -0,0 +1,40 @@
+use App\Entity\OrderItem;
+use Doctrine\ORM\EntityManagerInterface;
+
+class OrderCleaner
```
Потрібно спростити, перенести функціонал на сторону бази даних(set null, та cascade delete). Залишитися має лише видалення ордерів

vitaliysheverov commented on src/Services/Client/ClientFactory.php:
```diff
@@ -0,0 +1,52 @@
+{
+    private \Traversable $paymentMethods;
+
+    public function __construct(\Traversable $paymentMethods) {
```
Різний стиль PSR, { згідно твого стилю має бути рядком нижче

vitaliysheverov commented on src/Services/Client/ClientFactory.php:
```diff
@@ -0,0 +1,52 @@
+        $this->paymentMethods = $paymentMethods;
+    }
+
+    public function createClient(bool $card = false): Client
```
Два методи, уникнути булевний параметр

vitaliysheverov commented on tests/Unit/Services/Client/ClientFactoryTest.php:
```diff
@@ -0,0 +1,51 @@
+
+class ClientFactoryTest extends TestCase
+{
+    private Traversable $paymentMethods;
```
Недостатнє покриття

alex-belto replied:
Проблеми з сетерами, setName рахує як необроблене
коли зачетив дату в setCardExpirationDate в assert виявилось що там null хоча сетив з willreturn
якщо тут сетери не покривати то не впевнений що тут ще покрити можна

vitaliysheverov commented on src/Services/Payment/CardPaymentProcessor.php:
```diff
@@ -0,0 +1,44 @@
+            $this->em->getConnection()->beginTransaction();
+
+            $client->payOrder();
+            $client->setStatus(Client::ORDER_PAYED);
```
чому setStatus не інкапсульовний?

vitaliysheverov commented on tests/Unit/Services/Payment/CardPaymentProcessorTest.php:
```diff
@@ -0,0 +1,73 @@
+use Doctrine\ORM\EntityManagerInterface;
+use PHPUnit\Framework\TestCase;
+
+class CardPaymentProcessorTest extends TestCase
```
Непокритий сценарій catch при транзакції

vitaliysheverov commented on src/Services/Payment/CashPaymentProcessor.php:
```diff
@@ -0,0 +1,37 @@
+        try {
+            $this->em->getConnection()->beginTransaction();
+            $client->payOrder();
+            $client->setStatus(Client::ORDER_PAYED);
```
Те саме зі статусом, окрім цього, я б додав коментар, що явна транзакція в цих місцях використовується лише для демонстрації

vitaliysheverov commented on tests/Unit/Services/Payment/PaymentHandlerTest.php:
```diff
@@ -0,0 +1,119 @@
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\DependencyInjection\ServiceLocator;
+
+class PaymentHandlerTest extends TestCase
```
Недостатнє покриття, Присутній код, що не викликається в тестованому класі

alex-belto replied:
заблоковано enum
додати catch сценарій

vitaliysheverov commented on tests/Unit/Services/Restaurant/RestaurantBuilderTest.php:
```diff
@@ -0,0 +1,202 @@
+            ->method('findAll')
+            ->willReturn([$this->waiter, $this->waiter]);
+
+        $this->restaurant
```
чому немає with?

vitaliysheverov commented on tests/Unit/Services/Restaurant/RestaurantBuilderTest.php:
```diff
@@ -0,0 +1,202 @@
+            ->method('findAll')
+            ->willReturn([$this->waiter]);
+
+        $this->expectException(Exception::class);
```
Деталізувати excpetion(текст)

vitaliysheverov commented on src/Services/Restaurant/RestaurantBuilder.php:
```diff
@@ -0,0 +1,100 @@
+        return $restaurant;
+    }
+
+    public function reset(): Restaurant
```
чому ресет публічний? і для чого власне він потрібен, ти ж не тримаєш стейт ресторану в білдері

alex-belto replied:
на випадок якщо я виректора буду робити

vitaliysheverov commented on src/Services/Restaurant/RestaurantBuilder.php:
```diff
@@ -0,0 +1,100 @@
+    {
+        $waiters = $this->em->getRepository(Waiter::class)->findAll();
+        if (count($waiters) < $amount) {
+            throw new \Exception('U dont have enough staff in pull');
```
Текст краще звичайний, без сленгу

vitaliysheverov commented on src/Services/Restaurant/RestaurantBuilder.php:
```diff
@@ -0,0 +1,100 @@
+ * Constructs a restaurant by hiring staff,
+ * creating a menu for the restaurant, and retrieving the restaurant object.
+ */
+class RestaurantBuilder
```
Чому так багато публіних методів, які ніде не використовуються?

alex-belto replied:
для директора

vitaliysheverov commented on tests/Unit/Services/Restaurant/RestaurantManagerTest.php:
```diff
@@ -0,0 +1,52 @@
+        $em = $this->createMock(EntityManagerInterface::class);
+        $client = $this->createMock(Client::class);
+        $restaurant = $this->createMock(Restaurant::class);
+        $restaurant->method('getMaxVisitorsPerDay')->willReturn($maxVisitorPerDay);
```
чому немає expects?

vitaliysheverov commented on tests/Unit/Services/Restaurant/RestaurantProviderTest.php:
```diff
@@ -0,0 +1,84 @@
+        $this->restaurantBuilder = $this->createMock(RestaurantBuilder::class);
+        $this->em = $this->createMock(EntityManagerInterface::class);
+        $this->restaurantProvider = new RestaurantProvider($this->restaurantBuilder, $this->em);
+        $this->filePath = realpath(__DIR__ . '/../../../..') . $_ENV['FILE_PATH'];
```
це дуже небезпечно, чому аж на головний рівень виходиш?
Краще мати папку біля тестів для такого

alex-belto replied:
встановив змінну env для тестового оточення з шляхом до файла в пвпкі env
App\Tests\Unit\Services\Restaurant\RestaurantProviderTest::testRestaurantFileExist
TypeError: Mock_RestaurantBuilder_5663dc3c::buildRestaurant(): Argument #1 ($days) must be of type int, null given, called in /var/www/app/src/Services/Restaurant/RestaurantProvider.php on line 42
якщо передаю параметр то запускає білдер і свариться на те що він не мав би запускатись

alex-belto committed:
upd parameter name in StaffResolver.php - $randomStaffIdx

alex-belto committed:
delete unused var in index.php

alex-belto committed:
rename controller from RestaurantManager.php to RestaurantManagerController.php

alex-belto committed:
rename controller from RestaurantManager.php to RestaurantManagerController.php

alex-belto committed:
rename var in MenuFixtures.php

alex-belto committed:
removed unused dependency

alex-belto committed:
rename fixtures

alex-belto committed:
fix MenuFixturesTest.php because of remove dependency from MenuFixture

alex-belto committed:
rename isEnoughMoney method and fixed tests

alex-belto committed:
removed unused element from Order entity

alex-belto committed:
removed KitchenerOrderProcessor.php to KitchenerListener.php

alex-belto committed:
removed WaiterOrderProcessor.php to WaiterListener.php

alex-belto committed:
removed OrderManagerInterface.php

alex-belto committed:
remove ClientCleaner.php & OrderCleaner.php to DataCleaner.php

alex-belto committed:
removed ability to set null to client card data

alex-belto committed:
rename dish & drink to plural in RestaurantBuilder.php

alex-belto commented on src/Entity/MenuItem.php:
```diff
@@ -5,29 +5,45 @@
+    private ?Restaurant $restaurant = null;
 
-    public function getId(): ?int
+    #[ORM\ManyToOne(inversedBy: 'menuItem')]
```
один menuItem може належати багатьом OrderItem

alex-belto replied:
томущо багато orderItem можуть мати один menuItem

alex-belto committed:
specify the type for $cardNumber in ClientFactory.php

alex-belto committed:
remove transformation of type in ClientListener.php

alex-belto committed:
updated comment to classes

alex-belto commented on src/Services/Payment/TipsCardPaymentDecorator.php:
```diff
@@ -0,0 +1,35 @@
+     */
+    public function pay(Client $client, Order $order): void
+    {
+        $this->cardPaymentProcessor->pay($client, $order);
```
кількість чайових визначється рандомно в OrderListener
а саме списання чайових іде через $this->tipsDistributor->splitTips($client->getConnectedOrder());
дистрибьютор визначає стратегію чайових і вираховує з клієнта

alex-belto committed:
rewrite RestaurantBuilder class

alex-belto committed:
RestaurantManager.php check loop

alex-belto committed:
change bool param on method which return bool param

alex-belto commented on src/Services/Restaurant/RestaurantManager.php:
```diff
@@ -0,0 +1,75 @@
+            for ($j = 1; $j <= $visitorsPerDay; $j++) {
+                $client = $this->clientFactory->createClient(true);
+                $this->em->persist($client);
+                $this->em->flush();
```
тут проблема в тому що коли ми записужмо клієнта відпрацьовує лісенер і ми не можемо тримати всіх разом на запис

alex-belto committed:
changed const on enum

alex-belto committed:
add expects to methods,
minor changes to var name

alex-belto commented on tests/Unit/Listeners/Client/ClientListenerTest.php:
```diff
@@ -0,0 +1,65 @@
+
+        $orderItemFactory
+            ->expects($this->exactly(3))
+            ->method('createOrderItem');
```
App\Tests\Unit\Listeners\Client\ClientListenerTest::testMakeOrder
Expectation failed for method name is "createOrderItem" when invoked 3 times
Parameter 1 for invocation App\Services\OrderItem\OrderItemFactory::createOrderItem(Mock_MenuItem_aaf109af Object (...), App\Entity\Order Object (...)): App\Entity\OrderItem does not match expected value.

 ->with($this->equalTo($menuItem), $this->equalTo($mockOrder)) ругається на мокнутий ордер, хоча параметр вказаний як обовязковий

alex-belto commented on tests/Unit/Listeners/Order/OrderListenerTest.php:
```diff
@@ -0,0 +1,50 @@
+        $this->orderListener->payOrder($this->order);
+    }
+
+    public function testOrderDone(): void
```
теж дивна річ 
$this->order
            ->expects($this->once())
            ->method('getClient')
            ->willReturn($this->client);
            мав би викликатися 1 раз, пише 0 викликів;
$this->orderListener->payOrder($this->client);
пише що очікує параметр ордер, хоча і оригінальному методі очікує клієнат
       public function payOrder(Client $client): void

alex-belto committed:
changed ClientFactory calss && updated ClientFactoryTest.php

alex-belto committed:
enum changes
remove getIndex method

alex-belto committed:
encapsulated setStatus from payment classes to method payOrder in Client entity

alex-belto committed:
add test to CardPaymentProcessorTest.php
testCatchException - skipped

alex-belto committed:
add text to expectExceptionMessage in RestaurantBuilderTest.php

alex-belto committed:
remove unused method from RestaurantManagerTest.php
and add expects to getDays method

alex-belto committed:
remove path to restaurant file from controller put it inside RestaurantProvider.php
add path to restaurant file for test env in phpunit.xml.dist
change path to restaurant file in .env

alex-belto commented on src/Entity/OrderItem.php:
```diff
@@ -0,0 +1,121 @@
+
+    public function __construct()
+    {
+        $this->menuItem = new ArrayCollection();
```
якщо наприклад нам треба буде дістати всі замовлення по одному блюду з меню
всі замолені картоплі фрі наприклад

alex-belto committed:
rewrite DataCleanerTest.php

alex-belto committed:
rewrite DataCleanerTest.php

alex-belto committed:
remove unused repository

alex-belto committed:
update ClientListenerTest.php

alex-belto committed:
add migration

alex-belto committed:
xdebug fix
client listener upd

alex-belto committed:
add 'with' to RestaurantBuilderTest.php
remove order return from ClientListenerTest.php

alex-belto committed:
updated ClientFactoryTest.php

alex-belto committed:
updated CardPaymentProcessorTest.php

alex-belto committed:
updated PaymentHandlerTest.php

alex-belto committed:
updated OrderTest && RestaurantProviderTest.phpranProvider

alex-belto committed:
updated RestaurantBuilder.php and tests to:
RestaurantProviderTest.php and
RestaurantBuilderTest.php

alex-belto committed:
updated RestaurantManagerTest.php

alex-belto committed:
updated RestaurantManagerControllerTest.php

alex-belto committed:
.idea

alex-belto committed:
merge

alex-belto committed:
merge & changes for merge
+ remove dynamic property php problem

alex-belto committed:
change the way env param use, now we use it by service parameters

alex-belto committed:
used fixture in RestaurantManagerControllerTest.php instead RestaurantProvider.php service
create fixture TestRestaurantFixture.php
change env var name from FILE_PATH to RESTAURANT_FILE_PATH
change RestaurantProvider.php method name from getPath to getRestaurantFilePath

alex-belto committed:
fixed looping bug
corrected tests to fix
add new status to order

alex-belto committed:
moved RestaurantFixtureTest
add fixture to test

alex-belto committed:
Merge pull request #21 from alex-belto/functional_tests

Functional tests
